Helper methods( Filtering and Updating) for Notification model

omarelkashef opened the pull request:
This PR adds helper methods  that facilitate provide important operations to the Notification model. That is, we can now get and filter by receiver which would be very useful for the inbox. Also, we can filter by unread, read, deleted, and the type of the notification and mark multiple notifications as deleted, undeleted, read, unread.

Commit:
added group invitation as an option for notification type

Commit:
added include_deleted param to filtering methods

Commit:
make include_deleted false by default

Commit:
fixed mistakes and added restore deleted method

Commit:
removed comment from another issue

AndrewM131 commented on src/chigame/users/models.py:
```diff
@@ -93,24 +93,68 @@ class GroupInvitation(models.Model):
+        self.update(visible=True)
+
+    def is_read(self):
+        return self.filter(read=True)
```
My only question is when is_read, is_unread, is_deleted, and is_not_deleted, how does the system know to only mark a specific notification as read, unread, deleted or not deleted? Does it need an additional input?

omarelkashef replied:
These methods are called on querysets so for example for is_read, the queryset will have Notification where read is True

AndrewM131 replied:
And does the filter individualize it so only one is affected, and not all?

omarelkashef replied:
All Notification that fit this criteria are included in the queryset.

AndrewM131 replied:
I see, thank you for the clarification. the helper methods look properly implemented in the Notification model.

AndrewM131 reviewed (approved):
LGTM, thank you for clarifying my questions.

elizabethli31 reviewed (approved):
Tested all methods, and everything works great. Very complete helpers as always. Good work! Update from dev and I'll merge

omarelkashef commented:
Closing statment:  In this pull request, we introduced helpful methods that streamline essential operations for the Notification model. With these additions, we can now retrieve and filter notifications by receiver. Additionally, we enabled filtering by unread, read, deleted, and notification type the ability as well as marking multiple notifications as deleted, undeleted, read, or unread. 

@24raniwalar worked on the mark_x() and is_x() methods
@omarelkashef  worked on the filter and get methods

elizabethli31 commented:
Issue Score: Excellent

Comments:
Nice set of functions, and really clean code. Great job!

<!-- {"score": "excellent", "minor": false} -->
